File: frontend/CodeInTasks/src/app/web-api/models/task/task-filter-model.ts
import { RunnerType, TaskCategory } from "../../enums";

export class TaskFilterModel {
    public categories: TaskCategory[] = [];
    public runners: RunnerType[] = [];

    public creatorId: string | null = null;

    public takeOffset: number = null!;
    public takeCount: number = null!;
}

File: backend/Shared/src/CodeInTasks.WebApi.Models/Task/TaskFilterModel.cs
namespace CodeInTasks.WebApi.Models.Task
{
    public class TaskFilterModel
    {
        [Required]
        public IEnumerable<TaskCategory> Categories { get; set; }

        [Required]
        public IEnumerable<RunnerType> Runners { get; set; }

        public Guid? CreatorId { get; set; }

        [Range(0, int.MaxValue)]
        public int TakeOffset { get; set; }

        [Range(1, ModelConstants.TaskFilter_TakeCount_Max)]
        public int TakeCount { get; set; }
    }
}
